Improve Overview page

r4victor opened the pull request:
This PR addresses Overview page issues described in #1374:

* Mentions that dstack takes care of provisioning compute resource, not only container orchestration.
* Expands "How does it work?" to make it more useful, better present dstack features, and prepare for the further reading of the docs.

r4victor committed:
Improve Overview page

peterschmidt85 commented on docs/docs/index.md:
```diff
@@ -1,13 +1,13 @@
-> well as on-premises clusters.
->
-> Offers native support for NVIDIA GPU and Google Cloud TPU accelerator chips.
+!!! info "Supported clouds"
```
"on-prem clusters" is a bit lost. people might think we only support clouds

peterschmidt85 commented on docs/docs/index.md:
```diff
@@ -19,15 +19,21 @@ It streamlines the management of dev environments, task execution on clusters, a
-1. Define configurations such as [dev environments](concepts/dev-environments.md), [tasks](concepts/tasks.md), 
+1. [Install](installation/index.md) `dstack`.
+   Configure the open-source server or sign up with [`dstack Sky`](https://sky.dstack.ai/).
+   With the open-source option, you host the server and bring your cloud credentials.
```
same: "on-prem clusters" is a bit lost. people might think we only support clouds.

peterschmidt85 reviewed (commented):
Suggested changes address:
1. Focus on both cloud and on-premises
2. Make `How does it work` even more informative and better structured
3. Removed `Why use dstack` (because of low value)
4. Removed `dstack Sky` mention from the middle of `How does it work` (to focus more on the open-source and not distract the user.

To be done outside of this PR:
- mention gateway and volume
- mention dstack Sky at the bottom

peterschmidt85 committed:
Suggested changes address:
1. Focus on both cloud and on-premises
2. Make `How does it work` even more informative and better structured
3. Removed `Why use dstack` (because of low value)
4. Removed `dstack Sky` mention from the middle of `How does it work` (to focus more on the open-source and not distract the user.

To be done outside of this PR:
- mention gateway and volume
- mention dstack Sky at the bottom

peterschmidt85 commented:
@r4victor please review the changes! I looks good, feel free to merge!
